app/models/user.rb

```diff
@@ -317,13 +317,18 @@ def self.find_or_create_external_user(attrs, auth_source_name)
     else
       User.as_anonymous_admin do
         auth_source = AuthSourceExternal.create!(:name => auth_source_name) if auth_source.nil?
-        user = User.create!(attrs.merge(:auth_source => auth_source))
-        user.locations = auth_source.locations
-        user.organizations = auth_source.organizations
-        if external_groups.present?
-          user.usergroups = auth_source.external_usergroups.where(:name => external_groups).map(&:usergroup).uniq
+        user = User.new(attrs.merge(:auth_source => auth_source))
+        if user.save
+          user.locations = auth_source.locations
+          user.organizations = auth_source.organizations
+          if external_groups.present?
+            user.usergroups = auth_source.external_usergroups.where(:name => external_groups).map(&:usergroup).uniq
+          end
+          user.post_successful_login
```

Comment:
It'd make sense to log that user has been created, we do it for ldap (see `User.try_to_auto_create_user`)
```suggestion
          logger.info  User '#{user.login}' auto-created from #{user.auth_source}"
          user.post_successful_login
```